scripts/check_crate_versions.sh

```
@@ -0,0 +1,54 @@
+MACROS_CRATE=""
+STD_CRATE=""
+PACKAGE_CRATES=()
+CONTRACT_CRATES=()
+
+check_crate_version() {
+    local cargo_toml=$1
+    local crate_name=$(grep '^name = ' $cargo_toml | cut -d'"' -f2)
+    local local_version=$(grep '^version = ' $cargo_toml | cut -d'"' -f2)
+    
+    # Get latest version from crates.io
+    local remote_version=$(cargo search $crate_name --limit 1 | grep "^$crate_name = " | cut -d'"' -f2 || echo "0.0.0")
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)